tx/tx_legacy.go

```diff
@@ -145,3 +86,15 @@ func (t *legacyTransaction) evaluateWork(origin thor.Address) func(nonce uint64)
 		return r.Div(math.MaxBig256, r)
 	}
 }
+
+// Below are the methods that are not compatible with legacy transaction
+func (t *legacyTransaction) maxFeePerGas() *big.Int         { return common.Big0 }
+func (t *legacyTransaction) maxPriorityFeePerGas() *big.Int { return common.Big0 }
+
+func (t *legacyTransaction) encode(*bytes.Buffer) error {
+	panic("encode called on LegacyTx")
```

Comment:
Should an error be returned instead ?

Reply:
No necessarily, panic is a sanity check in development stage, logic has already covered in transaction.go that this won't be called

Reply:
Unless someone changes something in the transaction.go 🙃  
But I get the idea. I was coming from the point of, if the method was returning an error, why not leverage it.